### DISCUSSION TITLE
Build output should reveal static prop size

### DISCUSSION THREAD

**Discussion by Timer** [vercel/next.js#62356] [Ideas]
Like we expose First Load JS size, we should expose the SSG data size for a page. This will help users know if they're including way too much data in their `getStaticProps` return value.

**Comment by joshhunt** [comment]
I was really susprised this was not the case already - I went searching this github for an issue for what I thought was a bug.

After I build my site, this is the output of the page sizes


```
Page                                                                                  Size     First Load JS
┌ ● /                                                                                 422 B          74.3 kB
├   └ css/7446edadc7c1dd7cbee8.css                                                    311 B
├   /_app                                                                             5.09 kB        62.8 kB
├ ○ /404                                                                              3.25 kB          66 kB
├ ● /version/[id]                                                                     400 B          74.3 kB
├   └ css/d4780b93ca0b227e8488.css                                                    275 B
├   ├ /version/ef65788d-4bda-4a32-9b89-e8d60bc00d7d
├   ├ /version/ecdd8569-3e81-4a2a-9c7e-029b24cb7335
├   ├ /version/fdddf2ca-57f5-4da0-88d9-10be10a553d5
├   └ [+6 more paths]
├ ● /version/[id]/[table]                                                             353 B           100 kB
├   ├ /version/ecdd8569-3e81-4a2a-9c7e-029b24cb7335/DestinyArtDyeReferenceDefinition
├   ├ /version/ecdd8569-3e81-4a2a-9c7e-029b24cb7335/DestinyUnlockDefinition
├   ├ /version/ecdd8569-3e81-4a2a-9c7e-029b24cb7335/DestinySandboxPerkDefinition
├   └ [+49 more paths]
└ ● /version/shadowkeep/DestinyInventoryItemDefinition                                370 B           100 kB
+ First Load JS shared by all                                                         62.8 kB
  ├ static/pages/_app.js                                                              5.09 kB
  ├ chunks/67a96815193a29f953673b8047b5767ebd2a0a96.e7100e.js                         10.7 kB
  ├ chunks/framework.c6faae.js                                                        40 kB
  ├ runtime/main.8ed1f4.js                                                            6.3 kB
  ├ runtime/webpack.c21266.js                                                         746 B
  └ css/d3fe8250de733785822c.css                                                      944 B
```

When I load `/version/shadowkeep/DestinyInventoryItemDefinition` directly in my browser, the page is 1.5 MB gzipped / 20 MB raw (this is massive, but expected)

![image](https://user-images.githubusercontent.com/46142/97294961-9fa37880-1846-11eb-9958-c52a1ae70625.png)

On the filesystem, the _nest/data .json file for the page is 16.6 MB

![image](https://user-images.githubusercontent.com/46142/97295275-fd37c500-1846-11eb-8e62-244c1699537b.png)

I would have expected at least the .html file size to be reflected in the page size output - I'm not sure what these numbers are that it's referencing. In my opinion, this output is bugged and shows inaccurate information. It tells me the above page is 370 B which is far from true :)

**Comment by lachlanjc** [comment]
![C0739F91-0BCE-4AD0-BBF5-8A15F6D61E25](https://user-images.githubusercontent.com/5074763/199646490-686ca399-5c4d-488d-a51b-4a01a4b69962.png)

These shipped at some point!

**Comment by samcx** [comment]
Moving this to <picture data-single-emoji=":nextjs:" title=":nextjs:"><img class="emoji" width="20" height="auto" src="https://emoji.slack-edge.com/T0CAQ00TU/nextjs/2a85d633ae594556.png" alt=":nextjs:" align="absmiddle"></picture> [discussions](https://github.com/vercel/next.js/discussions) since this is a feature request!
